### DISCUSSION TITLE
How to get the service instance port?

### DISCUSSION THREAD

**Discussion by xiaozaiyuji** [apache/skywalking#11443] [Q&A]
I have some question about the metric instance_traffic why this did not have the instance port information. This is not design for the instance_traffic  data?

**Comment by wu-sheng** [comment]
There isn't. A process doesn't have to have or/and only have one port.
So, there is no general property for port.

**Reply by xiaozaiyuji** [reply]
So there are any index store the instance port? I need this data to do some check,but I did not know the right index.Can you tell me ?

**Reply by wu-sheng** [reply]
No, there isn't.
Port only exists for a specific trace, when it is a RPC relative context. Nothing more.

**Reply by xiaozaiyuji** [reply]
Thanks for your help,I find the network_address_alias,does this will help me for some angle。I query the data find this contains port.

**Reply by wu-sheng** [reply]
Be careful, those are only IPs used at the RPC client side, and changing with network relationship changing.
I would say that is not a good data source.

**Reply by xiaozaiyuji** [reply]
Thank you so much, In fact ,I try to combine the ebpf data and skywalking data.But I have not enough knowledge about the skywalking index structure.So I find the data.I will take your advice and not take this data.I will split the ebpf data and skywalking data with different type.
